### PR TITLE
Archive.org Remove leading numeric

### PR DESCRIPTION
**Pull request by inverse**
Allows clean scrobbling for releases such as https://archive.org/details/AH003_corwin_trails_-_corwin_trails/Oasis_is_Myth.mp3

### CODE REVIEW

File: src/connectors/archive.js [hunk 1]
```diff
@@ -39,7 +39,29 @@ Connector.getArtistTrack = () => {
...
+
+function hasAllTracksNumericPrefix(trackSelector) {
+	const tracks = document.querySelectorAll(trackSelector);
+	if (tracks.length === 0) {
+		return false;
+	}
+
+	let hasAllTracksNumericPrefix = true;
+	const re = /^\d+\w+/;
+	for (let i = 0; i < tracks.length; ++i) {
```
Comment (alexesprit) [inline src/connectors/archive.js:59]: `tracks` is NodeList, which is iterable, so please consider using `for..of` instead.

File: src/connectors/archive.js [hunk 2]
```diff
@@ -39,7 +39,29 @@ Connector.getArtistTrack = () => {
...
+		if (!re.test(tracks[i].innerText)) {
+			hasAllTracksNumericPrefix = false;
```
Comment (alexesprit) [inline src/connectors/archive.js:61]: You could omit the temp variable and return the value directly.

**Review by alexesprit — CHANGES_REQUESTED** [review]

File: src/connectors/archive.js [hunk 3]
```diff
@@ -39,7 +39,29 @@ Connector.getArtistTrack = () => {
 		track = trackParts[2];
 	}
 
+	if (hasAllTracksNumericPrefix('.jwrowV2 .ttl')) {
+		track = track.trim().replace(/^(\d+\w+)/, '');
+	}
+
```
Comment (alexesprit) [inline src/connectors/archive.js:45]: Will this work if we move this code inside a MetadataFilter function?

File: src/connectors/archive.js [hunk 4]
```diff
@@ -39,7 +39,29 @@ Connector.getArtistTrack = () => {
...
+function hasAllTracksNumericPrefix(trackSelector) {
+	const tracks = document.querySelectorAll(trackSelector);
+	if (tracks.length === 0) {
+		return false;
+	}
+
+	let hasAllTracksNumericPrefix = true;
+	const re = /^\d+\w+/;
+	for (let i = 0; i < tracks.length; ++i) {
+		if (!re.test(tracks[i].innerText)) {
```
Comment (alexesprit) [inline src/connectors/archive.js:60]: Is there a reason to use `innerText` here?

**Comment by alexesprit** [comment]
Also I would include the link from PR description as a comment, like we do in the Bandcamp connector.

File: src/connectors/archive.js [hunk 5]
```diff
@@ -39,7 +39,29 @@ Connector.getArtistTrack = () => {
...
 Connector.albumSelector = '.thats-left > h1 [itemprop=name]';
+
+function hasAllTracksNumericPrefix(trackSelector) {
+	const tracks = document.querySelectorAll(trackSelector);
+	if (tracks.length === 0) {
+		return false;
+	}
+
+	let hasAllTracksNumericPrefix = true;
+	const re = /^\d+\w+/;
```
Comment (alexesprit) [inline src/connectors/archive.js:58]: I think this expression could be extracted to a global scope.

File: src/connectors/archive.js (hunk 1 above)
Comment (inverse) [reply src/connectors/archive.js:59]: Always forget this one

File: src/connectors/archive.js (hunk 2 above)
Comment (inverse) [reply src/connectors/archive.js:61]: I like to avoid multiple return points when possible.

File: src/connectors/archive.js (hunk 3 above)
Comment (inverse) [reply src/connectors/archive.js:45]: How would that work? what would it be like?

File: src/connectors/archive.js (hunk 4 above)
Comment (inverse) [reply src/connectors/archive.js:60]: What would you use? I wanted to get the text only of an element

**Comment by inverse** [comment]
> Also I would include the link from PR description as a comment, like we do in the Bandcamp connector.

Like a link to a page that does this?

File: src/connectors/archive.js (hunk 4 above)
Comment (alexesprit) [reply src/connectors/archive.js]: `textContent` is prefferable in most cases. See [this](https://stackoverflow.com/a/35213639) answer for details.

File: src/connectors/archive.js (hunk 3 above)
Comment (alexesprit) [reply src/connectors/archive.js:47]:
```js
const filter = new MetadataFilter({ track: removeNumericPrefixes });

Connector.applyFilter(filter);

function removeNumericPrefixes(track) {
	if (hasAllTracksNumericPrefix('.jwrowV2 .ttl')) {
		return track.trim().replace(/^(\d+\w+)/, '');
	}

	return track;
}
```

Also, it's worth using a constant for `.jwrowV2 .ttl` selector.

**Comment by alexesprit** [comment]
> Like a link to a page that does this?

Yes, it could be helpful for debugging the connector.

File: src/connectors/archive.js (hunk 3 above)
Comment (inverse) [reply src/connectors/archive.js:47]: Thanks - I'll give this a shot later :pray:

**Comment by inverse** [comment]
@alexesprit any other feedback?

**Comment by alexesprit** [comment]
@inverse it's almost perfect now, aside the `textContent` request change. :)

**Comment by inverse** [comment]
Sorry - forgot to push it :see_no_evil:

**Review by alexesprit — APPROVED** [review]

**Pull request merged**